javascript/ql/test/library-tests/frameworks/Spife/tests.ql

```diff
@@ -0,0 +1,254 @@
+import javascript
+import semmle.javascript.security.dataflow.CleartextStorageCustomizations
+import semmle.javascript.security.dataflow.CorsMisconfigurationForCredentialsCustomizations
+import semmle.javascript.security.dataflow.StackTraceExposureCustomizations
+import semmle.javascript.security.dataflow.ServerSideUrlRedirectCustomizations
+import semmle.javascript.security.dataflow.RequestForgeryCustomizations
+import semmle.javascript.security.dataflow.ReflectedXssCustomizations
+import semmle.javascript.security.dataflow.ReflectedXssQuery as XssConfig
+import semmle.javascript.heuristics.AdditionalRouteHandlers
+
```

Comment:
```suggestion

```

This was the reason that your tests "worked" before [the fix](https://github.com/github/codeql/commit/4ba3190d29f3b5273c38e8f93b59c42ee36ba480).  

Your `Spife::RouteSetup` and `Spife::StandardRouteHandler` were effectively unused in your tests before [the fix](https://github.com/github/codeql/commit/4ba3190d29f3b5273c38e8f93b59c42ee36ba480), because your tests don't distinguish between heuristic and non-heuristic route-handlers.  
You have to distinguish between heuristic and non-heuristic, otherwise you're only testing the former. 

You can try to delete `Spife::StandardRouteHandler`, your tests still pass as they are right now.  

You have a bunch of test failures when you delete this line, and I see three possible solution to that:   
- Accept the new test output (including failures), to document that some things can only be detected using the heuristic route-handlers.  
- Delete the tests lines are only detected by the heuristic route-handlers.   
   This is fine, our other heuristic route-handlers don't really have tests.  
- Add another test that tests the heuristic route-handlers. 

(The above assumes that no other bug surfaces from this).  

Sorry for figuring this out so late, but I didn't look that closely at your test implementations.